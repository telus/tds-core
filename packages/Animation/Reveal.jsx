import React from 'react'
import PropTypes from 'prop-types'
import { Transition } from 'react-transition-group'

const defaultStyle = (timeout, delay) => {
  return {
    transition: `height ${timeout}ms ${delay ? `${delay}ms` : ''}`,
    overflow: 'hidden',
  }
}

const transitionStyles = height => ({
  entering: { height: `${height}px` },
  entered: { height: `${height}px`, overflow: 'visible' },
  exiting: { height: 0 },
  exited: { height: 0, visibility: 'hidden' },
})

const Reveal = ({ duration, timeout, height, delay, children, ...rest }) => {
  const transitionTimeout = duration || timeout
  return (
    <Transition {...rest} timeout={transitionTimeout + delay}>
      {status => (
        <div
          style={{
            ...defaultStyle(transitionTimeout, delay),
            ...transitionStyles(height)[status],
          }}
          aria-hidden={status === 'exited'}
          data-testid="childrenContainer"
        >
          {children()}
        </div>
      )}
    </Transition>
  )
}
Reveal.propTypes = {
  timeout: PropTypes.number.isRequired,
  delay: PropTypes.number,
  duration: PropTypes.number,
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  children: PropTypes.func.isRequired,
}

Reveal.defaultProps = {
  duration: undefined,
  delay: undefined,
}

export default Reveal
