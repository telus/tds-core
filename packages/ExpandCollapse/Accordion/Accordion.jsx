import React from 'react'
import PropTypes from 'prop-types'
import { componentWithName } from '@tds/util-prop-types'

import Panels from '../Panels'
import Panel from '../Panel/Panel'

/**
 * A specialized `ExpandCollapse` in which only one panel can be open at a time.
 *
 * @version ../package.json
 */
class Accordion extends React.Component {
  state = {
    openPanel: this.props.open,
    prevOpenPanel: null,
  }

  static getDerivedStateFromProps(props, state) {
    const { prevOpenPanel } = state
    const open = props.open

    if (open !== prevOpenPanel) {
      return {
        openPanel: open,
        prevOpenPanel: open,
      }
    }
    return null
  }

  togglePanel = panelId => {
    const { onToggle } = this.props

    this.setState(
      ({ openPanel }) => {
        return {
          openPanel: openPanel === panelId ? undefined : panelId,
        }
      },
      () => {
        if (onToggle) {
          onToggle(this.state.openPanel)
        }
      }
    )
  }

  isPanelOpen = panelId => {
    return this.state.openPanel === panelId
  }

  render() {
    const { children, ...rest } = this.props

    return (
      <Panels {...rest} isPanelOpen={this.isPanelOpen} togglePanel={this.togglePanel}>
        {children}
      </Panels>
    )
  }
}

Accordion.propTypes = {
  /**
   * A panel identifier to programmatically open.
   */
  open: PropTypes.string,
  /**
   * Whether or not to show the divider above the first panel.
   */
  topDivider: PropTypes.bool,
  /**
   * A callback function to be invoked when any panel is opened or closed.
   *
   * @param {String} openPanel The currently open panel identifier.
   */
  onToggle: PropTypes.func,
  /**
   * Panel content. Can be text, any HTML element, or any component.
   * If content is not provided, only the Panel heading will be
   * displayed in a non-interactive manner. See documentation for usage.
   *
   * @since 2.1.0
   */
  children: componentWithName('Panel'),
}

Accordion.defaultProps = {
  open: undefined,
  topDivider: true,
  onToggle: undefined,
  children: undefined,
}

Accordion.Panel = Panel

export default Accordion
