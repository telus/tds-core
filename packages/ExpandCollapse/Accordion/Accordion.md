The `Accordion` has the same design and behaviours as the `ExpandCollpse`, except that only one panel can be opened at
a time.

Note that the `Accordion`'s `open` prop accepts only a single string, while the `ExpandCollapse`'s `open` prop accepts an
array of strings.

See [`ExpandCollapse`](#expandcollapse) above for examples.
