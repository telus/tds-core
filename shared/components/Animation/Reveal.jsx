import React from 'react'
import PropTypes from 'prop-types'
import { Transition } from 'react-transition-group'

const defaultStyle = (duration, timeout) => ({
  transition: `height ${duration || timeout}ms`,
  height: 0,
  overflow: 'hidden',
})

const hiddenContent = {
  visibility: 'hidden',
}

const transitionStyles = height => {
  const styles = {
    height: `${height}px`,
  }

  return {
    entering: styles,
    entered: { height: 'auto' },
    exiting: styles,
    exited: hiddenContent,
  }
}

const Reveal = ({ height, children, ...rest }) => (
  <Transition {...rest}>
    {status => (
      <div
        style={{
          ...defaultStyle(rest.duration, rest.timeout),
          ...transitionStyles(height)[status],
        }}
        aria-hidden={status === 'exited'}
        data-testid="childrenContainer"
      >
        {children()}
      </div>
    )}
  </Transition>
)
Reveal.propTypes = {
  timeout: PropTypes.number.isRequired,
  duration: PropTypes.number,
  height: PropTypes.number.isRequired,
  children: PropTypes.func.isRequired,
}

Reveal.defaultProps = {
  duration: undefined,
}

export default Reveal
