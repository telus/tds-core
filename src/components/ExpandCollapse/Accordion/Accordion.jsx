import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'

import Panels from '../Panels'
import Panel from '../Panel/Panel'

/**
 * A specialized `ExpandCollapse` in which only one panel can be open at a time.
 */
class Accordion extends React.Component {
  state = {
    openPanel: this.props.open,
  }

  componentWillReceiveProps(nextProps) {
    const nextPanel = nextProps.open

    if (this.state.openPanel !== nextPanel) {
      this.setState({ openPanel: nextPanel })
    }
  }

  togglePanel = panelId => {
    const { onToggle } = this.props

    this.setState(
      ({ openPanel }) => {
        return {
          openPanel: openPanel === panelId ? undefined : panelId,
        }
      },
      () => {
        if (onToggle) {
          onToggle(this.state.openPanel)
        }
      }
    )
  }

  isPanelOpen = panelId => {
    return this.state.openPanel === panelId
  }

  render() {
    const { children, ...rest } = this.props

    return (
      <Panels {...rest} isPanelOpen={this.isPanelOpen} togglePanel={this.togglePanel}>
        {children}
      </Panels>
    )
  }
}

Accordion.propTypes = {
  /**
   * A panel identifier to programmatically open.
   */
  open: PropTypes.string,
  /**
   * Whether or not to show the divider above the first panel.
   */
  topDivider: PropTypes.bool,
  /**
   * A callback function to be invoked when any panel is opened or closed.
   *
   * @param {String} openPanel The currently open panel identifier.
   */
  onToggle: PropTypes.func,
  /**
   * The expandable panels. Must be at least one `Accordion.Panel`.
   */
  children: childrenOfType(Panel).isRequired,
}

Accordion.defaultProps = {
  open: undefined,
  topDivider: true,
  onToggle: undefined,
}

Accordion.Panel = Panel

export default Accordion
