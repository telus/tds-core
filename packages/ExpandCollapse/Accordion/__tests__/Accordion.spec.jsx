import React from 'react'
import { shallow, mount } from 'enzyme'

import Reveal from '../../../../shared/components/Animation/Reveal'
import Accordion from '../Accordion'

describe('Accordion', () => {
  const doShallow = (props = {}) =>
    shallow(
      <Accordion {...props}>
        <Accordion.Panel id="the-panel" header="The header">
          The panel content
        </Accordion.Panel>
      </Accordion>
    ).dive()

  const doMount = component => {
    const accordion = mount(component)

    const findPanel = id => accordion.find(`[data-testid="${id}"]`)
    const findPanelHeader = id => findPanel(id).find('button')

    return {
      accordion,
      findPanel,
      findPanelHeader,
      hoverPanel: id => findPanelHeader(id).simulate('mouseEnter'),
      unHoverPanel: id => findPanelHeader(id).simulate('mouseLeave'),
      clickPanel: id => findPanelHeader(id).simulate('click'),
    }
  }

  const expectPanelToBeOpen = panel => expect(panel.find(Reveal)).toHaveProp('in', true)
  const expectPanelToBeClosed = panel => expect(panel.find(Reveal)).toHaveProp('in', false)

  const defaultProps = {
    id: 'a-panel',
    header: 'Panel title',
  }
  const defaultChildren = 'Panel content'
  const aPanel = ({ ...props }, children = defaultChildren) => (
    <Accordion.Panel {...defaultProps} {...props}>
      {children}
    </Accordion.Panel>
  )

  it('renders a closed accordion', () => {
    const { accordion } = doMount(
      <Accordion>
        <Accordion.Panel id="panel-1" header="Panel title">
          Panel content
        </Accordion.Panel>
      </Accordion>
    )

    expect(accordion).toMatchSnapshot()
  })

  it('renders an open accordion', () => {
    const { accordion } = doMount(
      <Accordion open="panel-1">
        <Accordion.Panel id="panel-1" header="Panel title">
          Panel content
        </Accordion.Panel>
      </Accordion>
    )

    expect(accordion).toMatchSnapshot()
  })

  it('can have 1 panel open by default', () => {
    const { findPanel } = doMount(
      <Accordion open="panel-1">
        {aPanel({ id: 'panel-1' })}
        {aPanel({ id: 'panel-2' })}
      </Accordion>
    )

    expectPanelToBeOpen(findPanel('panel-1'))
    expectPanelToBeClosed(findPanel('panel-2'))
  })

  it('can open and close a panel', () => {
    const { findPanel, clickPanel } = doMount(<Accordion>{aPanel({ id: 'panel-1' })}</Accordion>)

    clickPanel('panel-1')
    expectPanelToBeOpen(findPanel('panel-1'))

    clickPanel('panel-1')
    expectPanelToBeClosed(findPanel('panel-1'))
  })

  it('closes all other panels when opening a new panel', () => {
    const { findPanel, clickPanel } = doMount(
      <Accordion>
        {aPanel({ id: 'panel-1' })}
        {aPanel({ id: 'panel-2' })}
      </Accordion>
    )

    clickPanel('panel-1')
    expectPanelToBeOpen(findPanel('panel-1'))

    clickPanel('panel-2')
    expectPanelToBeClosed(findPanel('panel-1'))
    expectPanelToBeOpen(findPanel('panel-2'))
  })

  it('lets a parent component control the open panel', () => {
    const { accordion, findPanel } = doMount(
      <Accordion>
        {aPanel({ id: 'panel-1' })}
        {aPanel({ id: 'panel-2' })}
      </Accordion>
    )

    accordion.setProps({ open: 'panel-1' })
    expectPanelToBeOpen(findPanel('panel-1'))

    accordion.setProps({ open: 'panel-2' })
    expectPanelToBeClosed(findPanel('panel-1'))
    expectPanelToBeOpen(findPanel('panel-2'))
  })

  it('triggers a callback when any panel is opened or closed', () => {
    const onToggle = jest.fn()

    const { clickPanel } = doMount(
      <Accordion onToggle={onToggle}>{aPanel({ id: 'panel-1' })}</Accordion>
    )

    clickPanel('panel-1')
    expect(onToggle).toHaveBeenCalledWith('panel-1')
  })

  it('passes additional attributes to the element', () => {
    const accordion = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(accordion).toHaveProp('id', 'the-id')
    expect(accordion).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const accordion = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(accordion).not.toHaveProp('className', 'my-custom-class')
    expect(accordion).not.toHaveProp('style')
  })
})
