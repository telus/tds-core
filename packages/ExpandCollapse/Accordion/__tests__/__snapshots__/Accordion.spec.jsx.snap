// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders a closed accordion 1`] = `
<Accordion
  topDivider={true}
>
  <Panels
    isPanelOpen={[Function]}
    togglePanel={[Function]}
    topDivider={true}
  >
    <div
      className="base"
    >
      <HairlineDivider
        gradient={false}
        vertical={false}
      >
        <hr
          className="TDS_HairlineDivider-modules__horizontalThin___jAiIF TDS_HairlineDivider-modules__horizontal___14xom TDS_HairlineDivider-modules__reset___e92-M TDS_Spacing-modules__noSpacing___XPYDG TDS_Borders-modules__none___1fCjZ"
        />
      </HairlineDivider>
      <PanelWrapper
        key="panel-1"
        onClick={[Function]}
        open={false}
        panelDisabled={false}
        panelHeader="Panel title"
        panelId="panel-1"
      >
        <div
          data-testid="panel-1"
        >
          <div>
            <Clickable
              aria-expanded={false}
              dangerouslyAddClassName="header"
              disabled={false}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              type="button"
            >
              <button
                aria-expanded={false}
                className="clickable header"
                disabled={false}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                type="button"
              >
                <Box
                  inline={false}
                  tag="div"
                  vertical={3}
                >
                  <div
                    className="TDS_Box-modules__verticalPadding-3___Fsv37"
                  >
                    <Box
                      between={3}
                      inline={true}
                      tag="div"
                    >
                      <div
                        className="TDS_Box-modules__betweenRightMargin-3___1dOvx TDS_Box-modules__inline___jTHcz"
                      >
                        <div>
                          <Translate
                            direction="y"
                            distance="0.25rem"
                            in={false}
                            timeout={300}
                          >
                            <Transition
                              appear={false}
                              enter={true}
                              exit={true}
                              in={false}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={300}
                              unmountOnExit={false}
                            >
                              <div
                                style={
                                  Object {
                                    "transition": "transform 300ms",
                                  }
                                }
                              >
                                <Text
                                  block={false}
                                  bold={false}
                                  invert={false}
                                  size="large"
                                >
                                  <span
                                    className="TDS_Typography-modules__large___3Z5sJ TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__largeFont___3fZLf TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__color___1Jt_W"
                                  >
                                    <DecorativeIcon
                                      size={16}
                                      symbol="caretDown"
                                      variant="primary"
                                    >
                                      <Icon
                                        aria-hidden="true"
                                        size={16}
                                        symbol="caretDown"
                                        variant="primary"
                                      >
                                        <i
                                          aria-hidden="true"
                                          className="TDS_Icon-modules__iconCaretDown___3by8b TDS_Icon-modules__icon___13xYd TDS_Icon-modules__primary___2UY-N TDS_Icon-modules__size16___1pm12"
                                        />
                                      </Icon>
                                    </DecorativeIcon>
                                  </span>
                                </Text>
                              </div>
                            </Transition>
                          </Translate>
                        </div>
                        <Flexbox
                          dangerouslyAddClassName="headerAlign"
                          direction="row"
                        >
                          <div
                            className="row headerAlign"
                          >
                            <Flexbox
                              dangerouslyAddClassName="headerAlign"
                              direction="column"
                            >
                              <div
                                className="column headerAlign"
                              >
                                <div
                                  className="fullWidth"
                                >
                                  <Text
                                    block={false}
                                    bold={false}
                                    invert={false}
                                    size="large"
                                  >
                                    <span
                                      className="TDS_Typography-modules__large___3Z5sJ TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__largeFont___3fZLf TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__color___1Jt_W"
                                    >
                                      Panel title
                                    </span>
                                  </Text>
                                </div>
                              </div>
                            </Flexbox>
                          </div>
                        </Flexbox>
                      </div>
                    </Box>
                  </div>
                </Box>
              </button>
            </Clickable>
            <Reveal
              duration={500}
              height={0}
              in={false}
              timeout={0}
            >
              <Transition
                appear={false}
                duration={500}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={0}
                unmountOnExit={false}
              >
                <div
                  aria-hidden={true}
                  data-testid="childrenContainer"
                  style={
                    Object {
                      "height": 0,
                      "overflow": "hidden",
                      "transition": "height 500ms",
                      "visibility": "hidden",
                    }
                  }
                >
                  <div
                    data-testid="content"
                  >
                    <DimpleDivider>
                      <hr
                        className="TDS_DimpleDivider-modules__dimple___1OWjL TDS_Spacing-modules__noSpacing___XPYDG TDS_Borders-modules__none___1fCjZ"
                      />
                    </DimpleDivider>
                    <Box
                      inline={false}
                      inset={4}
                      tag="div"
                    >
                      <div
                        className="TDS_Box-modules__horizontalPadding-4___3W1-P TDS_Box-modules__verticalPadding-4___QMNNa"
                      >
                        <Text
                          block={true}
                          bold={false}
                          invert={false}
                          size="base"
                        >
                          <span
                            className="TDS_Typography-modules__base___gCTVZ TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__baseFont___2g1ZX TDS_Typography-modules__color___1Jt_W TDS_Typography-modules__blockText___2DsZN"
                          >
                            <Panel
                              disabled={false}
                              header="Panel title"
                              id="panel-1"
                              key=".0"
                            >
                              <div>
                                Panel content
                              </div>
                            </Panel>
                          </span>
                        </Text>
                      </div>
                    </Box>
                  </div>
                </div>
              </Transition>
            </Reveal>
            <HairlineDivider
              gradient={false}
              vertical={false}
            >
              <hr
                className="TDS_HairlineDivider-modules__horizontalThin___jAiIF TDS_HairlineDivider-modules__horizontal___14xom TDS_HairlineDivider-modules__reset___e92-M TDS_Spacing-modules__noSpacing___XPYDG TDS_Borders-modules__none___1fCjZ"
              />
            </HairlineDivider>
          </div>
        </div>
      </PanelWrapper>
    </div>
  </Panels>
</Accordion>
`;

exports[`Accordion renders an open accordion 1`] = `
<Accordion
  open="panel-1"
  topDivider={true}
>
  <Panels
    isPanelOpen={[Function]}
    open="panel-1"
    togglePanel={[Function]}
    topDivider={true}
  >
    <div
      className="base"
      open="panel-1"
    >
      <HairlineDivider
        gradient={false}
        vertical={false}
      >
        <hr
          className="TDS_HairlineDivider-modules__horizontalThin___jAiIF TDS_HairlineDivider-modules__horizontal___14xom TDS_HairlineDivider-modules__reset___e92-M TDS_Spacing-modules__noSpacing___XPYDG TDS_Borders-modules__none___1fCjZ"
        />
      </HairlineDivider>
      <PanelWrapper
        key="panel-1"
        onClick={[Function]}
        open={true}
        panelDisabled={false}
        panelHeader="Panel title"
        panelId="panel-1"
      >
        <div
          data-testid="panel-1"
        >
          <div>
            <Clickable
              aria-expanded={true}
              dangerouslyAddClassName="header"
              disabled={false}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              type="button"
            >
              <button
                aria-expanded={true}
                className="clickable header"
                disabled={false}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                type="button"
              >
                <Box
                  inline={false}
                  tag="div"
                  vertical={3}
                >
                  <div
                    className="TDS_Box-modules__verticalPadding-3___Fsv37"
                  >
                    <Box
                      between={3}
                      inline={true}
                      tag="div"
                    >
                      <div
                        className="TDS_Box-modules__betweenRightMargin-3___1dOvx TDS_Box-modules__inline___jTHcz"
                      >
                        <div>
                          <Translate
                            direction="y"
                            distance="-0.25rem"
                            in={false}
                            timeout={300}
                          >
                            <Transition
                              appear={false}
                              enter={true}
                              exit={true}
                              in={false}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={300}
                              unmountOnExit={false}
                            >
                              <div
                                style={
                                  Object {
                                    "transition": "transform 300ms",
                                  }
                                }
                              >
                                <Text
                                  block={false}
                                  bold={false}
                                  invert={false}
                                  size="large"
                                >
                                  <span
                                    className="TDS_Typography-modules__large___3Z5sJ TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__largeFont___3fZLf TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__color___1Jt_W"
                                  >
                                    <DecorativeIcon
                                      size={16}
                                      symbol="caretUp"
                                      variant="primary"
                                    >
                                      <Icon
                                        aria-hidden="true"
                                        size={16}
                                        symbol="caretUp"
                                        variant="primary"
                                      >
                                        <i
                                          aria-hidden="true"
                                          className="TDS_Icon-modules__iconCaretUp___3PL_K TDS_Icon-modules__icon___13xYd TDS_Icon-modules__primary___2UY-N TDS_Icon-modules__size16___1pm12"
                                        />
                                      </Icon>
                                    </DecorativeIcon>
                                  </span>
                                </Text>
                              </div>
                            </Transition>
                          </Translate>
                        </div>
                        <Flexbox
                          dangerouslyAddClassName="headerAlign"
                          direction="row"
                        >
                          <div
                            className="row headerAlign"
                          >
                            <Flexbox
                              dangerouslyAddClassName="headerAlign"
                              direction="column"
                            >
                              <div
                                className="column headerAlign"
                              >
                                <div
                                  className="fullWidth"
                                >
                                  <Text
                                    block={false}
                                    bold={false}
                                    invert={false}
                                    size="large"
                                  >
                                    <span
                                      className="TDS_Typography-modules__large___3Z5sJ TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__largeFont___3fZLf TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__color___1Jt_W"
                                    >
                                      Panel title
                                    </span>
                                  </Text>
                                </div>
                              </div>
                            </Flexbox>
                          </div>
                        </Flexbox>
                      </div>
                    </Box>
                  </div>
                </Box>
              </button>
            </Clickable>
            <Reveal
              duration={500}
              height={0}
              in={true}
              timeout={500}
            >
              <Transition
                appear={false}
                duration={500}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={500}
                unmountOnExit={false}
              >
                <div
                  aria-hidden={false}
                  data-testid="childrenContainer"
                  style={
                    Object {
                      "height": "auto",
                      "overflow": "hidden",
                      "transition": "height 500ms",
                    }
                  }
                >
                  <div
                    data-testid="content"
                  >
                    <DimpleDivider>
                      <hr
                        className="TDS_DimpleDivider-modules__dimple___1OWjL TDS_Spacing-modules__noSpacing___XPYDG TDS_Borders-modules__none___1fCjZ"
                      />
                    </DimpleDivider>
                    <Box
                      inline={false}
                      inset={4}
                      tag="div"
                    >
                      <div
                        className="TDS_Box-modules__horizontalPadding-4___3W1-P TDS_Box-modules__verticalPadding-4___QMNNa"
                      >
                        <Text
                          block={true}
                          bold={false}
                          invert={false}
                          size="base"
                        >
                          <span
                            className="TDS_Typography-modules__base___gCTVZ TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__baseFont___2g1ZX TDS_Typography-modules__color___1Jt_W TDS_Typography-modules__blockText___2DsZN"
                          >
                            <Panel
                              disabled={false}
                              header="Panel title"
                              id="panel-1"
                              key=".0"
                            >
                              <div>
                                Panel content
                              </div>
                            </Panel>
                          </span>
                        </Text>
                      </div>
                    </Box>
                  </div>
                </div>
              </Transition>
            </Reveal>
            <HairlineDivider
              gradient={false}
              vertical={false}
            >
              <hr
                className="TDS_HairlineDivider-modules__horizontalThin___jAiIF TDS_HairlineDivider-modules__horizontal___14xom TDS_HairlineDivider-modules__reset___e92-M TDS_Spacing-modules__noSpacing___XPYDG TDS_Borders-modules__none___1fCjZ"
              />
            </HairlineDivider>
          </div>
        </div>
      </PanelWrapper>
    </div>
  </Panels>
</Accordion>
`;
